### ISSUE TITLE
tslint "eofline" gives a warning, even though vim does add a newline to the end of the file

### ISSUE THREAD

**Issue by stroiman** [mhartington/nvim-typescript#191]
First of all, thanks for this plugin. I just tried installing it yesterday, and so far it seems to be working much more smoothly than my previous setup (typescript-language-server/NeovimLanguageClient) 

However,

# Problems summary

When adding `tslint-language-service` typescript compiler plugin, I get tslint warnings in vim (it seems like it doesn't appear without this plugin)

However, I get warnings for missing EOL in the last line, if I enable the `eofline` linter rule.

But neovim does add an EOL to the last line by default so this warning shouldn't appear

## Expected

No warning for missing EOL after the last line

## Environment Information

 * terminal: iTerm 3.2.3 (os-x mojave)
 * vim/nvim version: NVIM v0.3.1

## Provide a minimal vim rc with less than 50 lines (Required!)

```vim
call plug#begin()
Plug 'HerringtonDarkholme/yats.vim'
Plug 'mhartington/nvim-typescript', {'do': './install.sh'}
call plug#end()
```

Project's `tsconfig.json`

```json
{
    "compilerOptions": {
        "target": "es6",
        "module": "commonjs",
        "removeComments": false,
        "preserveConstEnums": true,
        "sourceMap": true,
        "outDir": "dist",
        "types": [],
        "plugins": [
          { "name": "tslint-language-service" }
        ]
    }
}
```

Project's `tslint.json`

```json
{
    "extends": [
        "tslint:recommended"
    ],
    "rules": {
      "eofline": true
    }
}
```

## The reproduce ways from neovim starting

From my example repo:
1. `npm install`
2. `./start index.ts` (./start adds node_modules/.bin to path, and start nvim with specific init file).
3. Edit the file, e.g. remove last semicolon and add it again. (It appears I need to do a change to the file to trigger the compilation?)
4. Show the location list `:lop`

## Example project repo?

https://github.com/stroiman/nvim-typescript-bug

## Screen shot (if possible)

![screenshot 2018-10-11 at 10 36 18](https://user-images.githubusercontent.com/237227/46791483-8c95dc80-cd41-11e8-8e2f-face8404daaa.png)

**Comment by mhartington** [comment]
This isn't an issue with this plugin. Since the error is coming from tslint-language-service, it's most likely related to TSLint, not neovim. Please direct the issue there.

**Comment by stroiman** [comment]
Yet, the tslint-launguage-server was also used by my previous setup, typescript-language-server/NeovimLanguageClient - and in that configuration it didn't exhibit those problems.

**Comment by mhartington** [comment]
So Im not seeing a new line be added by neovim as described in the original post.

**Comment by mhartington** [comment]
Ok, so this _should_ be accounted for. I've tested it out locally for a bit, and I think it handles EOL in nvim.

**Comment by stroiman** [comment]
The issue seems to be fixed after I updated my vim plugins 👍
